components/locid/src/extensions/mod.rs

```diff
@@ -178,6 +178,24 @@ impl Extensions {
             other,
         })
     }
+
+    pub(crate) fn iter_subtags(&self) -> impl Iterator<Item = &str> {
+        self.other
+            .iter()
+            .filter(|o| o.get_ext() <= 't')
+            .map(|o| o.iter_subtags())
+            .flatten()
+            .chain(self.transform.iter_subtags())
+            .chain(self.unicode.iter_subtags())
+            .chain(
+                self.other
+                    .iter()
+                    .filter(|o| o.get_ext() > 't')
+                    .map(|o| o.iter_subtags())
+                    .flatten(),
+            )
```

Comment:
`question`: can we use the enum of extension types here rather than stringified ASCII idx matching? It seems fragile.

Reply:
The ASCII idx matching is only for `other` and I think it is the only way to do it. The stringify impl, below this one, does the same thing.

Reply:
Why not `.filter(|o| [ExtensionType::Other].contains(o))`?

Reply:
I'm now using the exact pre-existing `Writeable` code.

Reply:
I misunderstood this code. I assumed it is looking for a particular type, while it was about alphabetic order.